endpoints/authentication/src/main.cpp

```diff
@@ -355,7 +347,7 @@ namespace irods::http::handler
 
 					auto bearer_token = irods::http::process_stash::insert(authenticated_client_info{
 						.auth_scheme = authorization_scheme::openid_connect,
-						.username = std::move(irods_name),
+						.username = *std::move(irods_username),
```

Comment:
Shouldn't this be `std::move(*irods_username))`?

Reply:
`*std::move(irods_username)` 'steals' the value contained in the optional. Doing this the other way leaves an optional that claims to hold something, but the value is missing.

Doing it this way should make potential future errors more obvious.

Reply:
But... that transfers ownership of the optional rather than the string, right?

Once we move the string, does the string in the optional matter?
How does `*std::move(irods_username)` result in the string inside of the optional moving into `.username`?

Reply:
Found it! See comment in example on this page: https://en.cppreference.com/w/cpp/utility/optional/operator*

There may be other places in the code that may do it the way that I did. Please open an issue for verifying/updating the code to do it in the way you've specified.

Reply:
Yep, the example on the operator* page is where I saw it.

>  How does *std::move(irods_username) result in the string inside of the optional moving into .username?

I'm pretty sure the idea is to create an 'rvalue' of the optional to move the data from the original to it. Then, you dereference the newly created rvalue to get the string, which should still be a move, since there's likely an rvalue optimization for it.

> There may be other places in the code that may do it the way that I did. Please open an issue for verifying/updating the code to do it in the way you've specified.

Will do.

Reply:
Done.